[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d

dzdang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75305
* __->__ #75261
* #75304
* #72353

Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431831](https://our.internmc.facebook.com/intern/diff/D35431831)

dzdang committed:
[Quant][core][bc-breaking][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d

Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.
In addition, quantized_max_pool1d has
been removed from the frontend, which was previously not used anyways.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35402484).

dzdang committed:
Update on "[Quant][core][bc-breaking][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.
In addition, quantized_max_pool1d has
been removed from the frontend, which was previously not used anyways.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Differential Revision: [D35402484](https://our.internmc.facebook.com/intern/diff/D35402484)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][bc-breaking][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.
In addition, quantized_max_pool1d has
been removed from the frontend, which was previously not used anyways.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Differential Revision: [D35402484](https://our.internmc.facebook.com/intern/diff/D35402484)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][bc-breaking][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.
In addition, quantized_max_pool1d has
been removed from the frontend, which was previously not used anyways.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Differential Revision: [D35402484](https://our.internmc.facebook.com/intern/diff/D35402484)

[ghstack-poisoned]

jerryzh168 commented on test/forward_backward_compatibility/check_forward_backward_compatibility.py:
```diff
@@ -112,6 +112,7 @@
     ("aten::scatter_reduce.two", datetime.date(2022, 4, 15)),
     ("aten::_s_where", datetime.date(2022, 9, 30)),
     ("aten::quantized_max_pool2d", datetime.date(2022, 4, 15)),
+    ("aten::quantized_max_pool1d", datetime.date(2022, 4, 15)),
```
ideally these deprecations should be seaparate PR so that it's easy to revert in case something goes wrong

dzdang committed:
Update on "[Quant][core][bc-breaking][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.
In addition, quantized_max_pool1d has
been removed from the frontend, which was previously not used anyways.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.



[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][bc-breaking][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.
In addition, quantized_max_pool1d has
been removed from the frontend, which was previously not used anyways.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][bc-breaking][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][bc-breaking][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431831).

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431831](https://our.internmc.facebook.com/intern/diff/D35431831)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431831](https://our.internmc.facebook.com/intern/diff/D35431831)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431831](https://our.internmc.facebook.com/intern/diff/D35431831)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431831](https://our.internmc.facebook.com/intern/diff/D35431831)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431831).

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431831](https://our.internmc.facebook.com/intern/diff/D35431831)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431831).

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431831](https://our.internmc.facebook.com/intern/diff/D35431831)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431831).

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431831](https://our.internmc.facebook.com/intern/diff/D35431831)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431831).

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431831](https://our.internmc.facebook.com/intern/diff/D35431831)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][core][improvements] Combined dispatch registration for max_pool1d & quantized_max_pool1d"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR removes the is_quantized check from max_pool1d and modifies
max_pool1d_impl to be compatible with int tensors.

This PR relies on https://github.com/pytorch/pytorch/pull/74560, which introduces
structured kernel support for quantized tensors and https://github.com/pytorch/pytorch/pull/72353.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431831](https://our.internmc.facebook.com/intern/diff/D35431831)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431831).
